pkg/controllers/management/clusterstats/statsaggregator.go

```diff
@@ -225,3 +225,12 @@ func (s *StatsAggregator) machineChanged(key string, machine *v3.Node) (runtime.
 	}
 	return nil, nil
 }
+
+func isNotTainted(m *v3.Node) bool {
+	for _, taint := range m.Spec.InternalNodeSpec.Taints {
+		if taint.Value == "true" && (taint.Effect == "NoSchedule" || taint.Effect == "NoExecute") {
```

Comment:
I was curious about the `true` part here, and when I looked at k3s and I don't actually see a taint on any of the nodes.

Reply:
what kind of nodes were they? 
I've seen this added to nodes: 
```
"taints": [
{
"effect": "NoExecute",
"key": "node-role.kubernetes.io/etcd",
"type": "/v3/schemas/taint",
"value": "true"
}
],
```
and this one as well
```
"taints": [
{
"effect": "NoSchedule",
"key": "node-role.kubernetes.io/controlplane",
"type": "/v3/schemas/taint",
"value": "true"
}
],
```

Reply:
I assumed that with a value true or false, then that means you could have a taint, but disable it. If that is not the case, then the remove check could be removed

Reply:
I spun up a k3d cluster with 2 workers and all 3 of the nodes don't have any taints. Are your taints from RKE cluster or k3s ?

Reply:
According to https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#taint-v1-core, `value` is not necessarily required and based on https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/, removing a taint is equivalent to setting the `value` to empty. Maybe the check here should instead be:
```suggestion
		if taint.Value != "" && (taint.Effect == "NoSchedule" || taint.Effect == "NoExecute") {
```

Reply:
Taints are complicated. Like labels, they have a key, value, and effect. Kubeadm for example sets
`node-role.kubernetes.io/master=:NoExecute` which is `{Key="node-role.kubernetes.io/master", Value="", Effect="NoSchedule"}`.  Removing this from the command line would NOT involve setting it to an empty string, instead you put a - at the end of the effect: `kubectl taint node node1 node-role.kubernetes.io/master=:NoExecute-`

A toleration for this would normally just use the Exists operator, as opposed for only tolerating a specific value:
```
  - key: "example-key"
    operator: "Exists"
    effect: "NoSchedule"
```